### PR TITLE
Observability overview tweaks

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

- Observability overview improvements after a discussion with @NHingerl 

**Related issue(s)**
See also #11128

### CODE REVIEW

File: docs/overview/51-telemetry-observability-overview.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: Telemetry and Observability
 type: Observability
 ---
 
-Out of the box, Kyma provides **telemetry** tools to collect and expose raw data, such as metrics, traces, and log data.
```
Comment (skhalash) [inline docs/overview/51-telemetry-observability-overview.md:6]: I think it's important to be consistent with the terms we use here:

* Telemetry is data produced by applications that we want to observe. Metrics, logs, and traces are different kinds of telemetry
* AFAIK there is no official term for software that exposes telemetry. `Fluent Bit` is a log collector and in `Prometheus` world it's called an exporter.
* Observability is an ability to understand an application's behavior based on the telemetry it produces.

I like the term `Observability solution` used further in the docs ;)

File: docs/overview/51-telemetry-observability-overview.md [hunk 2]
```diff
@@ -3,19 +3,19 @@ title: Telemetry and Observability
...
+Out of the box, Kyma provides tools to collect and expose **telemetry** data, such as metrics, traces, and log data.
 
 Of course, you'll want to view and analyse the data you're collecting. This is where **observability** comes in.
```
Comment (skhalash) [inline docs/overview/51-telemetry-observability-overview.md:8]: I am not sure about the last sentence: This is where **observability** comes in. Maybe something about **observability tools** instead, @NHingerl?

File: docs/overview/51-telemetry-observability-overview.md (hunk 1 above)
Comment (NHingerl) [reply docs/overview/51-telemetry-observability-overview.md:6]: I agree! I like your suggested text.

File: docs/overview/51-telemetry-observability-overview.md (hunk 2 above)
Comment (NHingerl) [inline docs/overview/51-telemetry-observability-overview.md:8]:
```suggestion
Of course, you'll want to view and analyse the data you're collecting. This is where **observability** tools come in. 
```

**Review by NHingerl — CHANGES_REQUESTED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**